semgrep-core/src/analyzing/CFG_build.ml

```diff
@@ -255,6 +264,32 @@ let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
       | None -> state.g |> add_arc (newi, state.exiti)
       | Some catchesi -> state.g |> add_arc (newi, catchesi));
       CfgFirstLast (newi, None)
+  (* Any DefStmts which are FuncDefs are reified as proper Func nodes
+     within the CFG, which contain their own smaller CFGs.
+  *)
+  | FuncStmt { fdef; ent; body } -> (
+      let cfg = cfg_of_stmts lang body in
+      let newi = state.g#add_node { F.n = NFunc { fdef; cfg; ent } } in
+      (* If there is no previous node, we add an edge from the entrance node.
+         Why? This is so that the dataflow can run on unreachable functions.
```

Comment:
great comment. Might be useful to copy it next to enteri, so better understand the need for enteri.